src/Agent.Worker/Release/ReleaseServer.cs

```diff
@@ -32,5 +34,15 @@ public IEnumerable<AgentArtifactDefinition> GetReleaseArtifactsFromService(int r
             var artifacts = _releaseHttpClient.GetAgentArtifactDefinitionsAsync(_projectId, releaseId, cancellationToken: cancellationToken).Result;
             return artifacts;
         }
+
+        public async Task<RMContracts.Release> UpdateReleaseName(
```

Comment:
UpdateRelease would require permission to manage releases

When agent runs this command, I believe it runs under the Build Service identity context. I don't think this identity has permission to update. We need to make change in RM service to grant this permission for this identity.